carapace-server/src/test/java/org/carapaceproxy/RawClientTest.java

```diff
@@ -565,6 +565,49 @@ public void testRequestsReadTimeout() throws Exception {
         }
     }
 
+    @Test
+    public void testClientsIdleTimeout() throws Exception {
+        stubFor(get(urlEqualTo("/index.html"))
+                .willReturn(aResponse()
+                        .withStatus(200)
+                        .withHeader("Content-Type", "text/html")
+                        .withHeader("Content-Length", "it <b>works</b> !!".length() + "")
+                        .withBody("it <b>works</b> !!")));
+
+        TestEndpointMapper mapper = new TestEndpointMapper("localhost", wireMockRule.port());
+        EndpointKey key = new EndpointKey("localhost", wireMockRule.port());
+
+        ConnectionsManagerStats stats;
+        try (HttpProxyServer server = HttpProxyServer.buildForTests("localhost", 0, mapper, tmpDir.newFolder());) {
+            server.start();
+            int port = server.getLocalPort();
+            assertTrue(port > 0);
+
+            RuntimeServerConfiguration currentConfiguration = server.getCurrentConfiguration();
+            currentConfiguration.setClientsIdleTimeoutSeconds(30);
+            server.getConnectionsManager().applyNewConfiguration(currentConfiguration);
+
+            stats = server.getConnectionsManager().getStats();
+
+            try (RawHttpClient client = new RawHttpClient("localhost", port)) {
+                for (int j = 0; j < 2; j++) {
+                    RawHttpClient.HttpResponse res = client.get("/index.html");
+                    Thread.sleep(60_000);
```

Comment:
maybe it is better to change the new parameter with a lower value and reduce test duration?